jfr-daemon/src/main/java/com/newrelic/jfr/daemon/JFRDaemon.java

```diff
@@ -40,11 +36,15 @@
 
   public static void main(String[] args) {
     try {
-      DaemonConfig config = buildConfig();
-      MBeanServerConnection mBeanServerConnection =
+      var config = buildConfig();
+      var mBeanServerConnection =
           new MBeanServerConnector(config).getConnection();
-      Optional<String> entityGuid = new RemoteEntityGuid(mBeanServerConnection).queryFromJmx();
-      var uploader = buildUploader(config, entityGuid);
+      var readinessCheck = new AtomicBoolean(false);
+
+      var entityGuid = new RemoteEntityGuid(mBeanServerConnection).queryFromJmx();
+      readinessCheck.set(true);
```

Comment:
When the fetching goes into a background thread, this setting will need to happen there.